docs/DeveloperGuide.md

```diff
@@ -314,59 +386,131 @@ where the lessons are isolated to the student for more personalised teaching.<br
+
+Given below is an example usage scenario and how viewing tag is executed:
+- **Step 1:** The user launches the application. The `Model` is initialized with the saved data (or sample data if there were no saved data). Tags from each person is loaded into `UniqueTagList` and `tagCounter` will store the corresponding number of students labelled with the tags.
+- **Step 2:** The user enter the command `tag` to view all tags. `Logic` calls `AddressBookParser` to parse this command string, creating a `TagCommand`.
+- **Step 3:** `Logic` executes the `TagCommand`. During execution, `TagCommand#execute()` instantiates a `CommandResult` with the `DisplayType` of `TAGS` as a signal for `MainWindow` to switch the center panel to show the tag list.
+- **Step 4:** `MainWindow` then handles this command by calling `CenterPanel#displayTagListPanel()` to display the tag list to the user.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** Tags with duplicate case-insensitive tag names for a person is not allowed. If the user tries to adds a tag with the same tag name to the person already with that tag, the new tag will not be added and `tagCounter` will not increment the count for this tag.<br></div>
+
+Figure I.4.1 shows a sequence diagram of how the tag list is displayed to the user with UI components.<br>
+<img src="images/ViewTagSequenceDiagramUi.png" width="800" />
+
+*Figure I.4.1: View tags UI sequence diagram*
+
+#### Design considerations
+**Aspect: Data Structures to support lesson operations**
+
+* **Alternative 1 (current implementation):** Use a `UniqueTagList` to store the tags created and a class field `tagCounter` to map each unique `Tag` to the number of persons labelled under it.
+  - Pros:
+    - Quicker retrieval and update of data using a `HashMap` for `tagCounter`.
+  - Cons:
+    - Each `Person` object has its own set of `Tags` which may be repetitive and memory-consuming if there is a large number of same tags.
+    - Retrieval of all tags and calculation of the number of persons labelled under each tag during the initialization of the application requires iterating through all persons in TAB.
+
+* **Alternative 2:** Each tag stores a list of persons or number of persons labelled with that tag.
+  - Pros:
+    - Faster retrieval of the number of persons under each tag.
+  - Cons:
+    - This could result in circular dependency since a `Person` keeps reference of a set of `Tags` and a `Tag` has to keep a reference to a list of `Persons` simultaneously.
+    - Updating the tags labelled for a `Person` requires modification of the data fields of the `Person`. Since TAB objects are immutable, this means that new copies of `Person` and `Tag` have to be created after every command that modifies the data. This could slow down the application when there is a large amount of data stored.
+  
+
+### Switching between students, calendar, and tags
```

Comment:
Actually if we have time should expand more on this for cases of FliteredList and how we retain the filtered list with `CommandResult` etc.

Reply:
no time xd